src/InsertionsClient/Models/Asset.cs

```diff
@@ -14,13 +14,13 @@ public sealed class Asset
         /// NuGet name.
         /// </summary>
         [DataMember(Name = "name", IsRequired = true)]
-        public string Name { get; set; }
+        public string? Name { get; set; }

```

Comment:
The data member is "required" so conceptually the property shouldn't need to be nullable, is that correct? Wondering if this wouldn't be better:

        [DataMember(Name = "name", IsRequired = true)]
        public string Name { get; set; } = string.Empty;

The `= string.Empty` satisfies the compiler but doesn't actually incur any run-time cost as the deserializer doesn't invoke constructors.

Same for the rest of the `DataMember` changes in the PR. Having the props types as not-nullable is more appropriate and saves from explicit null checks or bangs.

Reply:
Never mind, it is required but the value may still be `null`. Looks good!

Reply:
That makes a lot of sense. I was already looking for an excuse to make this non-nullable, but didn't notice "IsRequired". I will change it.

Reply:
Sorry for randomizing you, if the data is deserialized from JSON it can still be explicitly `null`. For example:

```
{"name: null, "version": "1"}
```

Assuming non-null for required members may be possible only for data deserialized from XML.

Reply:
I made a small test and confirmed that it can indeed be null despite having `IsRequired`.
I will leave this as it is.

Reply:
One more comment and then I'll shut up :) Of course, you can still keep the null-ness encapsulated in these classes if it makes working with them easier. For this particular prop you could do:

```
        [DataMember(Name = "name", IsRequired = true)]
        private string? _name;
        public string Name => _name ?? string.Empty;
```

Not sure if it's worth it, though.

Reply:
Comments are always welcome :) 
I guess that is a way we can achieve what we want. But I won't make any more changes to not spend unnecessarily long time on this and also for the sake of simplicity.